### ISSUE TITLE
Support rolling update of pods on migration

### ISSUE THREAD

**Issue by simenl** [FoundationDB/fdb-kubernetes-operator#400]
A pod update that requires migration, e.g. any changes to volumes, requires a migration. A migration requires a full set of the updated pods to run concurrently with the previous set of pods. Often this requires double the cpu, memory, and storage.

In a resource constrained cluster, these updates may not be possible to perform, due to the increase in resource required.
An option to do a rolling update would make this possible, given that the cluster can support the surge of one extra pod.

**Comment by brownleej** [comment]
Yeah, this seems like a reasonable feature. Maybe we could implement this as a limit on how many pods we can replace at once.

**Comment by johscheuer** [comment]
Like proposed in #404 it makes sense to add `maxSurge` which would behave like [maxSurge](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-surge) of a `Deployment`.
